task-launcher/src/tasks/shared/helpers/fetchAndParseCorpus.js

```diff
@@ -81,19 +79,39 @@ const transformCSV = (csvInput, numOfPracticeTrials, sequentialStimulus) => {
       currPracticeAmount = 0;
     }
 
-    // Only push in the specified amount of practice trials
-    if (newRow.notes !== 'practice') {
-      stimulusData.push(newRow);
-    } else if (newRow.notes === 'practice' && currPracticeAmount < numOfPracticeTrials) {
+    if (newRow.trialType === "instructions") {
+      if (task == "trog") { 
+        instructionData.push(newRow);
+      } else { 
+        stimulusData.push(newRow);
+      }
+      totalInstruction += 1;
+      totalTrials += 1;
+
+    } else if (newRow.notes === 'practice') { 
+      // Only push in the specified amount of practice trials
+      if (currPracticeAmount < numOfPracticeTrials) {
+        if (task === "trog") { 
+          practiceData.push(newRow);
+        } else { 
+          stimulusData.push.push(newRow);
+        }
+        currPracticeAmount += 1;
+        totalPractice += 1;
+        totalTrials += 1;
+      } else {
+        skippedPractice += 1;
+      }
+
+    } else if (newRow.notes !== 'practice') {
+      // everything else is stimulus
```

Comment:
As stated in my previous comments, can you please avoid changes to our parsing logic? Our code base is very integrated and almost all tasks rely on this code. I am hesitant to merge anything that affects all our tasks. I will look into our parsing logic soon.

Reply:
@tonyamurray , the reason we don't want to sort our corpora to put all practice trials at the beginning is that we have some tasks (e.g. math) which have different types of practice trials for different subtasks (e.g., subtraction, addition). the preferred behavior is to put all practice trials in the corpus where they are meant to be shown (e.g., at the beginning)